sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1479,6 +1479,42 @@ class Analyzer(
       // Skip the having clause here, this will be handled in ResolveAggregateFunctions.
       case h: UnresolvedHaving => h
 
+      case a: Aggregate =>
+        val resolvedGroupingExprs =
+          a.groupingExpressions.map(resolveExpressionTopDown(_, a))
+            .map(CleanupAliases.trimStructFieldAlias)
+
+        val resolvedAggExprs = a.aggregateExpressions
+          .map(resolveExpressionTopDown(_, a))
+          .map {
+            // Here we want to trim StructField Alias in complex expression
+            // Only Alias(GetStructField, name) can't be trimmed since we need NamedExpression.
+            case alias@Alias(_: GetStructField, _) => alias
+            case e => CleanupAliases.trimStructFieldAlias(e).asInstanceOf[NamedExpression]
```

Comment:
why do we need to do it?

Reply:
> why do we need to do it?

As i have said in comment. If not get rid of that case will got error 
```
 SPARK-31670: Resolve Struct Field in Grouping Aggregate with same ExprId *** FAILED *** (185 milliseconds)
[info]   java.lang.ClassCastException: org.apache.spark.sql.catalyst.expressions.aggregate.AggregateExpression cannot be cast to org.apache.spark.sql.catalyst.expressions.NamedExpression
[info]
```